File: concurrency_paralellism/concurrency/hello_word.java
class HelloThread extends Thread {
    public void run() {
        for (int i = 0; i < 5; i++) {
            System.out.println("Hello from thread! " + i);
            try {
                Thread.sleep(500);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }
    }
}

public class hello_word {
    public static void main(String[] args) {
        HelloThread thread = new HelloThread();
        thread.start();

        for (int i = 0; i < 3; i++) {
            System.out.println("Hello from main! " + i);
            try {
                Thread.sleep(1000);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
        }

        try {
            thread.join();
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}
 

File: concurrency_paralellism/concurrency/hello_word.py
import threading
import time 

def hello():
		for i in range(50):
			print("hello from therad!",i)
			time.sleep(0.00005)
thread= threading.Thread(target=hello)
thread.start()
for i in range(20):
    print("Hello from main!", i)
    time.sleep(0.0001)

thread.join()
